trim.go

```diff
@@ -0,0 +1,90 @@
+package gophplib
+
+import (
+	"fmt"
+	"strings"
+)
+
+type toStringAble interface {
+	toString() string
+}
+
+// Trim is a ported function that works exactly the same as PHP 5.6's trim
+// function. For more information, see the [official PHP documentation].
+//
+// In PHP 5.6, when attempting to use the trim() function with a data type other than a string,
+// it automatically converts the requested variable into a string before performing the trim.
+// To achieve the same behavior in Go, this function converts the requested data types into strings
+// and then utilize the trim function from the package strings.
+// For more detailed information about the trim function in the package strings, see the [strings's trim documentation]
+//
+// Reference:
+//   - https://www.php.net/manual/en/function.trim.php
+//   - https://github.com/php/php-src/blob/4b8f72da5dfb201af4e82dee960261d8657e414f/ext/standard/string.c#L840-L850
+//   - https://github.com/php/php-src/blob/4b8f72da5dfb201af4e82dee960261d8657e414f/Zend/zend_API.c#L425-L470
+//   - https://github.com/php/php-src/blob/4b8f72da5dfb201af4e82dee960261d8657e414f/Zend/zend_operators.c#L593-L661
+//   - https://github.com/php/php-src/blob/4b8f72da5dfb201af4e82dee960261d8657e414f/Zend/zend_API.c#L261-L302
+//
+// Test Cases:
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/strings/trim1.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/strings/trim.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/strings/trim_basic.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/strings/trim_variation1.phpt
+//
+// NOTE: This function does not support specifying a set of characters using string ranges (e.g., "a-z") in the chars parameter.
```

Comment:
```suggestion
// NOTE: This function does not support specifying a set of characters using string ranges (e.g., "a..z") in the chars parameter.
```

AFAIK it's `a..z` not `a-z`